File: SinaTopic.cpp
#include "iostream"
#include "vector"
#include "string"
#include "regex"
#include "set"
#include "map"
using namespace std;
string Process(string cs){
    string mod = "!@#$^&*()+-={}[];:,.<>?";
    for(auto var : mod){//删除多余符号
        cs.erase(std::remove(cs.begin(), cs.end(), var), cs.end());
    }      
    for(int j=0; j<cs.size(); j++){//所有字母小写
        if(cs[j]>='A' && cs[j]<='Z')
            cs[j] = cs[j] - 'A' + 'a';
    }
    cs[0] =cs[0]-'a'+'A';//首字母大写
    return cs;
}
int main(){
    int n; cin >> n; cin.ignore();
    std::map<string, int> topic;
    std::set<string> word;
    std::string s;
    for(int i=0; i<n; i++){//对于每一句话
    //cout << i <<endl;
        getline(cin, s);
        word.clear();
        string w = "";
        bool flag = false;
        for(int k=0; k<s.size(); k++){//对于每个字符
            if(s[k] == '#'){
                if(k == s.size()-1) break;//容错处理
                k++;
                w = "";
                while(s[k] != '#'){
                    w += s[k];
                    k++;
                }
                w = Process(w);
                //cout << "--------" << w << "--------------" <<endl;
                word.insert(w);
                if(k != s.size()-1) k++;
            }
        }
        for(auto var : word){//对每个出现一次的单词进行计数
            topic[var]++;
        }
    }
    string res;
    int max = 0;
    int count = 1;
    for(auto it = topic.begin(); it!=topic.end(); it++){//进行判断
        //cout << it->first <<" "<< it->second << endl;
        if(max < it->second){
            max = it->second;
            res = it->first;
            count = 0;
        }else if(max = it->second){
            if(it->first < res)
                res = it->first;
            count++;
        }
    }
    cout << res << endl;
    cout << max << endl;
    if(count != 0){
        cout << "And " << count << " more ...";
    }
    
    return 0;
}

File: SinaTopic.py
import re
#输出被多条微博提到的话题，所以是存话题，用map
#里面的int村的是提到的次数
sr = r'!@#$%^&*()+-={}[];:,.<>?/'
n = int(input())
topic = {}
p = r"#.+#"#什么正则表达式
pattern = re.compile(p)
for i in range(n):
    s = input()
    word = pattern.findall(s)[0].split("# #")#格式化
    for j in range(len(word)):
        for k in sr:
            word[j] = word[j].replace(k,'')#去掉所有符号
        word[j] = word[j].lower()#转换成小写
        word[j] = word[j].capitalize()#转换成首字母大写
    word = list(set(word))#去重
    for w in word:
        if w in topic:
            topic[w] = topic[w]+1
        else:
            topic[w] = 1
#第一行输出被最多条微博提到的话题，因此遍历找
# 第二行输出其被提到的微博条数。打印值
# 如果这样的话题不唯一，则输出按字母序最小的话题，进行比较和加1
# 并在第三行输出And k more ...，
# 其中k是另外几条热门话题的条数。输入保证至少存在一条话题。
s = ''
max = 0
count = 1
for key,val in topic.items():
    if max < val:
        max = val
        s = key
        cout = 1
    elif max == val:
        if key < s:
            s = key
        cout = cout + 1
print(s)
print(max)
if cout != 1:
    print("And %d more..." % (count))
